package/MDAnalysis/analysis/utils/autocorrelation.py

```diff
@@ -0,0 +1,155 @@
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+from __future__ import absolute_import, division
+import numpy as np
+from copy import deepcopy
+
+
+def autocorrelation(list_of_sets, tau_max, window_step=1):
+    r"""The descrete implementation of the autocorrelation function.
+
+    .. math::
+       C_{HB}^c(\tau) = \frac{\sum_{ij}h_{ij}(t_0)h'_{ij}(t_0+\tau)}{\sum_{ij}h_{ij}(t_0)}
+
+    Parameters
+    ----------
+    list_of_sets : list
+      List of sets,
+    tau_max : int
+      The last tau (inclusive) for which to carry out autocorrelation.
+    window_step : int, optional
+      The step for the t0 to perform autocorrelation (without the overlap). Default is 1.
+
+    Returns
+    --------
+    tau_timeseries : list of int
+        the tau for which the autocorrelation was calculated
+    timeseries : list of int
+        the autocorelation values for each of the tau values
+    timeseries_data : list of list of int
+        the raw data from which the autocorrelation is computed. The time dependant evolution can be investigated.
+
+    .. versionadded:: 0.19.2
+    """
+
+    # Check types
+    assert type(list_of_sets) == list, "list_of_sets must be a one-dimensional list of sets"
+    assert type(list_of_sets[0]) == set,  "list_of_sets must be a one-dimensional list of sets"
+
+    # Check dimensions of parameters
+    assert len(list_of_sets) > tau_max, "tau_max cannot be greater than the length of list_of_sets"
+
+    tau_timeseries = list(range(1, tau_max + 1))
+    timeseries_data = [[] for _ in range(tau_max)]
+
+    # calculate autocorrelation
+    for t in range(0, len(list_of_sets), window_step):
+        Nt = len(list_of_sets[t])
+
+        if Nt == 0:
+            continue
+
+        for tau in tau_timeseries:
+            if t + tau >= len(list_of_sets):
+                break
+
+            # continuous: IDs that survive from t to t + tau and at every frame in between
+            Ntau = len(set.intersection(*list_of_sets[t:t + tau + 1]))
+            timeseries_data[tau - 1].append(Ntau / float(Nt))
+
+    timeseries = [np.mean(x) for x in timeseries_data]
+
+    # at time 0 the value has to be one
+    tau_timeseries.insert(0, 0)
+    timeseries.insert(0, 1)
+
+    return tau_timeseries, timeseries, timeseries_data
+
+
+def correct_intermittency(list_of_sets, intermittency):
+    """
+    Process consecutive intermittency with a single pass over the data. The returned data can be used as input to
+    the function `autocorrelation` in order to calculate the discrete autocorrelation function with a given
+    intermittency.
```

Comment:
Can you add more text to explain what the function does? Equations, references, etc please.

Reply:
(I realize that you're not the original author of the function but it will now show up in a user-facing module so we need to be a bit clearer what to do with it.)

Reply:
Hi @orbeckst , with regards to the intermittency, together with @p-j-smith we are the original authors. This feature was not available for SurvivalProbability before. We will provide more information and hopefully we'll soon write it up as well. Thanks